File: tests/comparisons/programs/coins.cpp
#include <iostream>
#include <vector>
int main(){
    std::vector<int> coins = {1, 2, 5};
    int amount = 11;
    std::vector<int> dp(amount+1, 99);
    dp[0] = 0;
    for(int c : coins)
        for(int v=c;v<=amount;v++)
            if(dp[v-c]+1 < dp[v]) dp[v] = dp[v-c]+1;
    std::cout << dp[amount] << "\n";
}

File: tests/comparisons/programs/coins.rs
fn main() {
    let coins: [usize; 3] = [1, 2, 5];
    let amount = 11usize;
    // 99 stands in for "no way to make this amount"
    let mut dp = [99u32; 12];
    dp[0] = 0;
    for c in coins.iter() {
        for v in *c..=amount {
            let candidate = dp[v - c] + 1;
            if candidate < dp[v] { dp[v] = candidate; }
        }
    }
    println!("{}", dp[amount]);
}
